pkg/clustermanager/cluster_manager.go

```diff
@@ -289,6 +302,23 @@ func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster,
 		logger.V(3).Info("External etcd is ready")
 	}
 
+	if features.IsActive(features.TaintsSupport()) {
+		cc, err := c.clusterClient.GetEksaCluster(ctx, workloadCluster, workloadCluster.Name)
```

Comment:
I think we can avoid this call
If you rebase with main, you'll have available `currentSpec` in this method